heidrun/nara_json.rb

```diff
@@ -0,0 +1,575 @@
+             :each => record.field('description', 'item | itemAv | fileUnit',
+                                   'coverageDates'),
+             :as => :dates do
+      # providedLabel dates.map { |d| make_date_provided_label(d) }
+      self.begin dates.map { |d| make_begin_date(d) }
+      self.end dates.map { |d| make_end_date(d) }
+    end
+
+    description record.field('description', 'item | itemAv | fileUnit',
+                             'scopeAndContentNote | generalNoteArray',
+                             'generalNote')
+                      .map { |d| make_description(d) }
+
+    # <extent>[VALUE]</extent>
+    extent record.field('description', 'item | itemAv | fileUnit', 'extent')
+
+    # <specificRecordsTypeArray>
+    #  <specificRecordsType>
+    #  <termName>[VALUE]</termName>
+    # </specificRecordsType>
+    # <specificRecordsTypeArray>"
+    dcformat record.field('description', 'item | itemAv | fileUnit',
+                          'specificRecordsTypeArray', 'specificRecordsType',
+                          'termName')
+
+    identifier record.field('description', 'item | itemAv | fileUnit',
+                            'variantControlNumberArray',
+                            'variantControlNumber')
+                     .map { |vcn| make_identifier(vcn) }
+
+    # <languageArray>
+    # <language>
+    # <termName>[VALUE]</termName>
+    # </language>
+    # </languageArray>
+    language :class => DPLA::MAP::Controlled::Language do
+
+      # FIXME:  shouldn't this be providedLabel?
+      prefLabel record.field('description', 'item | itemAv | fileUnit',
+                             'languageArray', 'language', 'termName')
+    end
+
+    # <geographicReferenceArray>
+    #  <geographicPlaceName>
+    #  <termName>[VALUE]</termName>
+    #  </geographicPlaceName>
+    # </geographicReferenceArray>
+    spatial :class => DPLA::MAP::Place do
+      providedLabel record.field('description', 'item | itemAv | fileUnit',
+                                 'geographicReferenceArray',
+                                 'geographicPlaceName', 'termName')
+    end
+
+    publisher :class => DPLA::MAP::Agent do
+      providedLabel record.field('description', 'item | itemAv | fileUnit',
+                                 'organizationalContributorArray |' \
+                                   'personalContributorArray')
+                          .map { |el| make_publisher(el) }
+    end
+
+    relation record.field('description', 'item | itemAv | fileUnit',
+                          'parentFileUnit')
+                   .map { |el| make_relation(el) }
+
+    rights record.field('description', 'item | itemAv | fileUnit',
+                        'useRestriction')
+                 .map { |el| make_rights(el) }
+
+    # <topicalSubjectArray>
+    # <topicalSubject>
+    # <termName>[VALUE]</termName>
+    # </topicalSubject>
+    # </topicalSubjectArray>
+    subject :class => DPLA::MAP::Concept do
```

Comment:
See [comment on `contributor`](https://github.com/dpla/heidrun-mappings/pull/14/files#r33056051) and [comment on `spatial`](https://github.com/dpla/heidrun-mappings/pull/14/files#r33057678).